Controllers/Services/EmployeeReconciliation/EmployeeBlackoutPeriodService.cs

```diff
@@ -0,0 +1,115 @@
+using NewJobSurveyAdmin.Models;
+using NewJobSurveyAdmin.Services.CallWeb;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace NewJobSurveyAdmin.Services
+{
+    public class EmployeeBlackoutPeriodService
+    {
+        private CallWebService callWeb;
+        private NewJobSurveyAdminContext context;
+        private LoggingService logger;
+        private EmployeeCreationService insertionService;
+
+        public EmployeeBlackoutPeriodService(
+            NewJobSurveyAdminContext context,
+            CallWebService callWeb,
+            LoggingService logger,
+            EmployeeCreationService insertionService
+        )
+        {
+            this.context = context;
+            this.callWeb = callWeb;
+            this.logger = logger;
+            this.insertionService = insertionService;
+        }
+
+        public async Task<Employee> UpdateBlackoutPeriod(
+            Employee employee,
+            SurveyDates dates
+        )
+        {
+            // Update employee dates.
+            employee.SetDates(dates);
+
+            // Update in CallWeb.
+            await callWeb.UpdateSurvey(employee);
+
+            // Create a new timeline entry.
+            employee.TimelineEntries.Add(new EmployeeTimelineEntry
+            {
+                EmployeeActionCode = EmployeeActionEnum.UpdateByTask.Code,
+                EmployeeStatusCode = employee.CurrentEmployeeStatusCode,
+                Comment = $"Blackout dates unset by script. New dates: " +
+                          $"InviteDate → {dates.InviteDate.ToString("yyyy-MM-dd")}, " +
+                          $"Reminder1Date → {dates.Reminder1Date.ToString("yyyy-MM-dd")}, " +
+                          $"Reminder2Date → {dates.Reminder2Date.ToString("yyyy-MM-dd")}, " +
+                          $"DeadlineDate → {dates.DeadlineDate.ToString("yyyy-MM-dd")}."
+            });
+            context.Entry(employee).State = EntityState.Modified;
+
+            // Save.
+            await context.SaveChangesAsync();
+
+            return employee;
+        }
+
+        public async Task<EmployeeTaskResult> UpdateBlackoutPeriods()
+        {
+            var updatedEmployeeList = new List<Employee>();
+            var exceptionList = new List<string>();
+
+            var blackoutPeriodSetting = await context.AdminSettings.FirstOrDefaultAsync(
+                i => i.Key.Equals(AdminSetting.IsBlackoutPeriod)
```

Comment:
IDE is complaining this isn't defined `IsBlackoutPeriod`. I see it *is* defined in another PR https://github.com/bcgov/NewJobSurveyAdmin/pull/65/files#diff-04a79ab9ce39fdab9b07a5be1d63e0119631e454485f036d1a08edf1bcbf0065R14 
I gather the intent here is to merge all three PRs sequentially so needing them to work as independent PRs is likely not worth emphasizing at this point.